$OpenBSD: patch-utils_ghc-pkg_Main_hs,v 1.7 2019/09/30 11:44:18 kili Exp $

Horrible hack to get rid of the annoying warnings when running
pkg_add -u. This means that you do NOT get any such warning when
running ghc-pkg unregister ... -v0 --force.

Index: utils/ghc-pkg/Main.hs
--- utils/ghc-pkg/Main.hs.orig
+++ utils/ghc-pkg/Main.hs
@@ -1479,7 +1479,7 @@ modifyPackage fn pkgarg verbosity my_flags force = do
             = display pkgid
         | otherwise = display pkgid ++ "@" ++ display (installedUnitId pkg)
         where pkgid = mungedId pkg
-  when (not (null newly_broken)) $
+  when (not (null newly_broken || (force == ForceAll && verbosity <= Silent))) $
       dieOrForceAll force ("unregistering would break the following packages: "
               ++ unwords (map displayQualPkgId newly_broken))
 
